src/libmusig2.h

```diff
@@ -10,166 +10,178 @@
+ *              : L:            Concatenation of x_only public keys of signers.
+ *              : pk_parity:    Parity of aggregate pubkey.
+ *              : nr_signers:   The number of signers.
+ *              : nr_messages:  The number of messages.
 * */
 typedef struct{
     secp256k1_context* ctx;
-    secp256k1_pubkey aggr_pk;
+    secp256k1_pubkey aggr_pubkey;
     secp256k1_pubkey **aggr_R_list;
-    int par_pk;
+    unsigned char *L;
+    int pk_parity;
     int nr_signers;
     int nr_messages;
-    unsigned char *L;
 }musig2_context;
 
-/** Struct      : musig2_context_sig
- *  Purpose     : Stores the parameters of musig2 signer.
- *  Parameters  : mc: a musig2_context object including the parameters of musig2.
- *              : comm_list: Batch commitment list of a signer.
- *              : keypair: Public and secret keys of signers.
- *              : state: The state of musig2.
+/** musig2_context_signer stores the parameters of musig2 signer.
+ *  Parameters  : mc:           a musig2_context object including the parameters of musig2.
+ *              : comm_list:    Batch commitment list of a signer.
+ *              : keypair:      Public and secret keys of signers.
+ *              : state:        The state of musig2.
  * */
 typedef struct {
     musig2_context mc;
     secp256k1_keypair **comm_list;
     secp256k1_keypair keypair;
     int state;
-}musig2_context_sig;
+}musig2_context_signer;
 
-/** Struct      : musig2_partial_signature
- *  Purpose     : Stores the parameters to aggregate a partial signature.
- *  Parameters  : sig: The partial signature of a signer.
- *              : R: The aggregated commitment of the signature.
+/** musig2_context_signature stores the parameters to aggregate a partial signature.
+ *  Parameters  : signature: The partial signature of a signer.
+ *              : R:         The aggregated commitment of the signature.
  * */
 typedef struct{
-    unsigned char sig[MUSIG2_PARSIG_BYTES];
+    unsigned char signature[MUSIG2_PARSIG_BYTES];
     secp256k1_xonly_pubkey R;
-}musig2_partial_signature;
-
-/*** Free memory allocated in MuSig2 context ***/
-void musig2_context_free(musig2_context *mc);
-
-/*** Free memory allocated in MuSig2 context ***/
-void musig2_context_sig_free(musig2_context_sig *mcs);
-
-/** Function    : musig2_init_signer
- *  Purpose     : Initializes a musig2 signer. Generates the keypair and creates a list of batch commitments for  signer.
- *                Returns 0 if signer initialisation fails, 1 otherwise.
- *  Parameters  : IN/OUT    : mcs: A musig2_context_sig object including parameters of musig2 signer.
- *              : IN        : ctx: A secp256k1_context object.
- *                          : nr_messages: The number of messages.
- * Returns      : 1/0.
+}musig2_context_signature;
+
+
+/** Initialize a musig2 signer.
+ *
+ *  Returns: 0 if signer initialisation fails, 1 otherwise.
+ *
+ *  In/Out: mcs:            a musig2_context_signer object
```

Comment:
Is the context in-out or just out? in-out notation is for when a variable is both read and modified. But in this function the context is not read, but simply initialised, right?